docs/tutorial/shell.md

```diff
@@ -0,0 +1,131 @@
+# Arr.ai shell basics
+
+This tutorial will provide a brief walkthrough of the features of arr.ai's
+interactive shell.
+
+## Starting the shell
+
+There are two ways to start the interactive. The first is to use the `arrai i`
+command.
+
+```bash
+$ arrai i
+@> _
+```
+
+Alternately, if the `ai` shortcut is set up correctly (see the main
+[README](../../README.md) for instructions), you can use it instead.
+
+```bash
+$ ai
+@> _
+```
+
+Once you see the `@>` prompt, you can start entering expressions or special
+commands as shown below.
+
+When you are finished using the arr.ai shell, type the exit character to return
+to the OS shell. In Unix-like environments, press Ctrl+D. The output will be as
+follows.
+
+```arrai
+@> exit
+$ _
+```
+
+In Windows, TODO.
+
+## Evaluating expressions
+
+To evaluate an expression in the shell, simply type it in and press enter.
+
+```arrai
+@> "Hello, world!"
+'Hello, world!'
+@> 6 * 9
+54
+@> //math.pi
+3.141592653589793
+```
+
+A complex expression may be entered over multiple lines. Arr.ai automatically
+detects when an expression isn't complete. (Don't worry if you don't understand
+the following expressions. They will make sense later in the tutorial.)
+
+```arrai
+@> (function1(a, b, c)
+ >  + function2(d, e, f)
+ >  + function3(x, y, z)
+ > ) / 3
+```
+
+Observe that the prompt changes from `@>` to `>` to indicate that it's awaiting
+further input. Here's a more cmomplex example.
+
+```arrai
+@> let v = (x: 1, y: 2, z: 3);
+ > let length = (v.x^2 + v.y^2 + v.z^2)^0.5;
+ > cond (
+ >     length > 1: "too big",
+ >     length < 1: "too small",
+ >     *: "just right"
+ > )
+```
+
+Caution: The current approach to detection uses some simple heuristics such as
+counting balanced bracketing. It will not always accurately detect that an
+expression is incomplete. The following example will fail.
+
+```arrai
+@> 1 /
+```
+
+## Global variables
+
+In order to set a global name, use the `/set` command as follows:
+
+```arrai
+@> /set x = 42
+42
+@> /set y = 54
+54
+@> x / y
+0.7777777777777778
+```
+
```

Comment:
Perhaps something like 

> For more details refer to [Set](https://github.com/arr-ai/arrai/blob/master/README.md#set)

Reply:
Agreed. After the docs are moved out of the main README (#186).

Reply:
👍